src/test/scala/fundamentals/level04/LogParserTest.scala

```diff
@@ -0,0 +1,114 @@
+package fundamentals.level04
+
+import org.scalactic.TypeCheckedTripleEquals
+import org.scalatest.FunSpec
+import LogParser._
+
+class LogParserTest extends FunSpec with TypeCheckedTripleEquals {
+
+  describe("parseLog") {
+
+    it("should return a KnownLog") {
+      val knownLog = parseLog("I,147,mice in the air")
+      assert(knownLog === ???)
+    }
+    
+    it("should return an UnknownLog") {
+      val unknownLog = parseLog("X blblbaaaaa")
+      assert(unknownLog === ???)
+    }
+
+  }
+  
+  describe("parseLogFile") {
+    
+    it("should return a List of LogMessage") {
+      val logMessages = parseLogFile("I,147,mice in the air\nX blblbaaaaa")
+      assert(logMessages === List(???, ???))
+    }
+    
+    it("should return an empty List") {
+      val logMessages = parseLogFile("")
+      assert(logMessages === Nil)
+    }
+    
+  }
+  
+  describe("getUnknowns") {
+    
+    it("should return the unknowns only") {
+      val known = parseLog("I,147,mice in the air")
```

Comment:
Had to use `parseLog` instead of just creating a `KnownLog(Info, 147, "mice in the air")` because `KnownLog` constructor doesn't yet exist.

Reply:
We could define the ADT in a supporting file and import it for the tests to compile. Once the guys write their own ADT, we could get them to switch the import to their ADT.